### PR TITLE
Create a simple CORS filter

### PR DESCRIPTION
**Pull request by chasegawa**

### CODE REVIEW

File: portlet-security-util/src/main/java/org/jasig/web/filter/SimpleCorsFilter.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+ *     
+ *     <!-- Comma separated methods allowed -->
+ *     <param-name>allowMethod</param-name>
+ *     <param-value>POST, GET</param-value>
+ *     
+ *     <!--Comma separated list of allowed header values -->
+ *     <param-name>allowHeaders</param-name>
+ *     <param-value>Origin, X-Requested-With, Content-Type, Accept</param-value>
+ *   </init-param>    
+ * </filter>
```
Comment (jameswennmacher) [inline portlet-security-util/src/main/java/org/jasig/web/filter/SimpleCorsFilter.java]: As is this is a straight-up servlet filter.  This means you have to list all the domains for all environments (dev, test, prod, etc.) in the portlet's web.xml or have a substitution process.  Instead I suggest implementing a Spring HandlerInterceptor.  This allows you to inject the values from a properties file using Spring.  See http://docs.spring.io/spring/docs/4.1.8.RELEASE/spring-framework-reference/htmlsingle/#mvc-handlermapping-interceptor.

Comment (chasegawa) [inline portlet-security-util/src/main/java/org/jasig/web/filter/SimpleCorsFilter.java]: I added documentation for instantiating this in Spring. By leaving it as a simple servlet filter, users have the ability to use it as such, or use it in a Spring managed way. I also added another @Component based class that extends the filter to allow for more current Spring ways of instantiation

File: portlet-security-util/src/main/java/org/jasig/web/filter/component/ComponentCorsFilter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.web.filter.component;
```
Comment (jameswennmacher) [inline portlet-security-util/src/main/java/org/jasig/web/filter/component/ComponentCorsFilter.java]: I suggest putting this class in org.jasig.springframework.security.filter

**Comment by jameswennmacher** [comment]
Love it.  +1

**Pull request merged**